liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -554,7 +554,12 @@ protected void doExecute() throws Exception {
         // all files in the configDirectory,
         // which is where the server.xml is located if a specific serverXmlFile
         // configuration parameter is not specified.
-        util.watchFiles(pom, outputDirectory, testOutputDirectory, executor, artifactPaths, serverXmlFile);
+        try {
+            util.watchFiles(pom, outputDirectory, testOutputDirectory, executor, artifactPaths, serverXmlFile);
+        } catch (PluginScenarioException e) { // this exception is caught when the server has been stopped by another process
+            log.info(e); 
+            System.exit(0); // enter shutdown hook 
```

Comment:
It could just call `return` here, since this is in the main `doExecute` method of the mojo.